include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -95,100 +256,179 @@ namespace sparrow
         [[nodiscard]] ArrowSchema& schema();
         [[nodiscard]] const ArrowSchema& schema() const;
 
-        [[nodiscard]] std::size_t
-        compute_buffer_size(buffer_type buffer_type, int64_t length, data_type data_type) const;
-    };
+        arrow_schema_private_data* get_schema_private_data();
+        arrow_array_private_data* get_array_private_data();
 
-    inline
-    void arrow_proxy::initialize_children()
-    {
-        m_children.clear();
-        m_children.reserve(static_cast<std::size_t>(array().n_children));
-        for (int64_t i = 0; i < array().n_children; ++i)
-        {
-            m_children.emplace_back(array().children[i], schema().children[i]);
-        }
-    }
+        [[nodiscard]] bool is_arrow_array_valid() const;
+        [[nodiscard]] bool is_arrow_schema_valid() const;
+        [[nodiscard]] bool is_proxy_valid() const;
+
+        static void swap(arrow_proxy& lhs, arrow_proxy& rhs);
```

Comment:
swap is usually implemented a non template method and a free function calling the method on one of its argument. This allows unqualified calls to swap relying on ADL in generic code:

```cpp
template <class T>
void do_something(T& lhs, T& rhs)
{
    // do different things
    using std::swap; // fallback if there is no overload of swap for T
   swap(lhs, rhs);
}
```